src/csharp/SDK/Transformation.cs

```diff
@@ -142,6 +142,129 @@ public void DepthImageToColorCamera(Image depth, Image transformed)
+                this.calibration.ColorCameraCalibration.ResolutionHeight,
+                this.calibration.ColorCameraCalibration.ResolutionWidth * sizeof(short))
+            {
+                DeviceTimestamp = depth.DeviceTimestamp,
+            };
+
+            int bytesPerPixel;
+            switch (custom.Format)
+            {
+                case ImageFormat.Custom8:
+                    bytesPerPixel = sizeof(byte);
+                    break;
+                case ImageFormat.Custom16:
+                    bytesPerPixel = sizeof(short);
+                    break;
+                default:
+                    throw new NotSupportedException("Failed to support this format of custom image!");
+            }
+
+            Image transformedCustom = new Image(
+                custom.Format,
+                this.calibration.ColorCameraCalibration.ResolutionWidth,
+                this.calibration.ColorCameraCalibration.ResolutionHeight,
+                this.calibration.ColorCameraCalibration.ResolutionWidth * bytesPerPixel)
+            {
+                DeviceTimestamp = custom.DeviceTimestamp,
+            };
+
+            this.DepthImageToColorCameraCustom(depth, custom, transformedDepth, transformedCustom, interpolationType, invalidCustomValue);
+
+            return (transformedDepth, transformedCustom);
+        }
+
+        /// <summary>
+        /// Transforms an Image and an Custom Image from the depth camera perspective to the color camera perspective.
+        /// </summary>
+        /// <param name="depth">Depth image to transform.</param>
+        /// <param name="custom">Custom image to transform.</param>
+        /// <param name="transformedDepth">An transformed deth image to hold the output.</param>
+        /// <param name="transformedCustom">An transformed custom image to hold the output.</param>
+        /// <param name="interpolationType">Parameter that controls how pixels in custom image should be interpolated when transformed to color camera space.</param>
+        /// <param name="invalidCustomValue">Defines the custom image pixel value that should be written to transformedCustom image in case the corresponding.</param>
+        /// <remarks>
+        /// The <paramref name="transformedDepth"/> Image must be of the resolution of the color camera, and
+        /// of the pixel format of the depth image.
+        /// The <paramref name="transformedCustom"/> Image must be of the resolution of the color camera, and
+        /// of the pixel format of the custom image.
+        /// </remarks>
+        public void DepthImageToColorCameraCustom(Image depth, Image custom, Image transformedDepth, Image transformedCustom, TransformationInterpolationType interpolationType, uint invalidCustomValue)
+        {
+            if (depth == null)
+            {
+                throw new ArgumentNullException(nameof(depth));
+            }
+
+            if (custom == null)
```

Comment:
>custom [](start = 16, length = 6)

suggestion, similar as you did for the function above, consider to check the custom.Format is either ImageFormat.Custom8 or ImageFormat.Custom16, or throw NotSupportedException("Failed to support this format of custom image!");
Also, consider to check the transformedCustom.Format as well

Reply:
I was added format check of each input/output custom image.
In addition, I was added check format of these images matches.
Thanks,